swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/images/AmazonImageHandler.kt

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.swabbie.aws.images
+
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.kork.core.RetrySupport
+import com.netflix.spinnaker.swabbie.*
+import com.netflix.spinnaker.swabbie.aws.instances.AmazonInstance
+import com.netflix.spinnaker.swabbie.aws.launchconfigurations.AmazonLaunchConfiguration
+import com.netflix.spinnaker.swabbie.echo.Notifier
+import com.netflix.spinnaker.swabbie.exclusions.ResourceExclusionPolicy
+import com.netflix.spinnaker.swabbie.model.*
+import com.netflix.spinnaker.swabbie.orca.OrcaJob
+import com.netflix.spinnaker.swabbie.orca.OrcaService
+import com.netflix.spinnaker.swabbie.orca.OrchestrationRequest
+import kotlinx.coroutines.experimental.Deferred
+import kotlinx.coroutines.experimental.async
+import kotlinx.coroutines.experimental.runBlocking
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.stereotype.Component
+import java.time.Clock
+import java.util.*
+import kotlin.system.measureTimeMillis
+
+@Component
+class AmazonImageHandler(
+  registry: Registry,
+  clock: Clock,
+  notifier: Notifier,
+  resourceTrackingRepository: ResourceTrackingRepository,
+  resourceOwnerResolver: ResourceOwnerResolver<AmazonImage>,
+  exclusionPolicies: List<ResourceExclusionPolicy>,
+  applicationEventPublisher: ApplicationEventPublisher,
+  lockingService: Optional<LockingService>,
+  retrySupport: RetrySupport,
+  private val rules: List<Rule<AmazonImage>>,
+  private val imageProvider: ResourceProvider<AmazonImage>,
+  private val instanceProvider: ResourceProvider<AmazonInstance>,
+  private val launchConfigurationProvider: ResourceProvider<AmazonLaunchConfiguration>,
+  private val accountProvider: AccountProvider,
+  private val orcaService: OrcaService
+) : AbstractResourceTypeHandler<AmazonImage>(
+  registry,
```

Comment:
weird spacing, should be indented one?

Reply:
yeah it is. Verified in the IDE it is on the same line as the constructor members.